components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/hashing/HashUtils.java

```diff
@@ -243,4 +248,71 @@ private static String convertByteToHex(byte[] byteData) {
 
         return sb.toString();
     }
+
+    public static boolean detectChangesInLibraries() throws HashingException {
+        boolean iseDetected = false;
+        try {
+            Map<String, String> storedHashes = loadLibFileHashes();
+            String libPath = GatewayCmdUtils.getCLILibPath();
+            for (Map.Entry<String, String> entry : storedHashes.entrySet()) {
+                String filePath = libPath + File.separator + entry.getKey() + GatewayCliConstants.EXTENSION_ZIP;
+                try {
+                    if (Files.exists(Paths.get(filePath))) {
+                        String checkSum = getFileCheckSum(filePath);
+                        if (!checkSum.equals(entry.getValue())) {
+                            logger.debug("Checksum difference detected for key : " + entry.getKey());
+                            storedHashes.put(entry.getKey(), checkSum);
+                            iseDetected = true;
+                        }
+                    }
+                } catch (NoSuchAlgorithmException | IOException e) {
+                    logger.error("Error while calculating check sum for path : " + filePath, e);
+                }
+            }
+            if(iseDetected) {
+                storeLibHashes(storedHashes);
+            } else {
+                logger.debug("No checksum difference detected in the lib directories");
+            }
+            return false;
+        } catch (IOException e) {
+            throw new HashingException("Error while gateway library change detection", e);
+        }
+    }
+
+    /**
+     * Loads the stored resource hashes
+     *
+     *
+     * @return a map with id to hash mapping loaded from the CLI temp
+     * @throws IOException error while loading the stored hashes
+     */
+    private static Map<String, String> loadLibFileHashes() throws IOException {
+        String content = GatewayCmdUtils.loadStoredLibHashes();
+        Map<String, String> hashes = new HashMap<>();
+        if (StringUtils.isNotEmpty(content)) {
+            ObjectMapper objectMapper = new ObjectMapper();
+            hashes = objectMapper.readValue(content, Map.class);
+        }
+        return hashes;
+    }
+
+    private static String getFileCheckSum(String filePath) throws NoSuchAlgorithmException, IOException {
+        MessageDigest md = MessageDigest.getInstance(MD5);
+        md.update(Files.readAllBytes(Paths.get(filePath)));
+        byte[] digest = md.digest();
+        return DatatypeConverter.printHexBinary(digest).toUpperCase();
+    }
+
+    /**
+     * Store the calculated hashes of gateway lib folders
```

Comment:
Method description ending without `.` causes checkstyle issues
```suggestion
     * Store the calculated hashes of gateway lib folders.
```